src/main/java/pl/allegro/tech/opel/OpelParser.java

```diff
@@ -230,6 +233,26 @@ Rule IntNumber() {
         );
     }
 
+    Rule ListInstantiation() {
+        return FirstOf(EmptyList(), NonEmptyList());
+
+    }
+
+    Rule EmptyList() {
+        return Sequence(
+                "[ ",
+                "] ",
+                push(nodeFactory.emptyListInstantiation()));
```

Comment:
I suggest to put this push at the beginning of Args() (the same way I did in Declarations()) so you don't need EmptyList()